python/GafferArnoldUITest/VisualiserAlgoTest.py

```diff
@@ -0,0 +1,133 @@
+##########################################################################
+#
+#  Copyright (c) 2020, Cinesite VFX Ltd. All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are
+#  met:
+#
+#      * Redistributions of source code must retain the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above
+#        copyright notice, this list of conditions and the following
+#        disclaimer in the documentation and/or other materials provided with
+#        the distribution.
+#
+#      * Neither the name of Cinesite VFX Ltd. nor the names of
+#        any other contributors to this software may be used to endorse or
+#        promote products derived from this software without specific prior
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import unittest
+
+import IECore
+import IECoreScene
+
+import GafferTest
+import GafferArnoldUI.Private.VisualiserAlgo as VisualiserAlgo
+
+class VisualiserAlgoTest( GafferTest.TestCase ) :
+
+	def testConformToOSLNetworkFull( self ) :
+
+		# Tests reserved word suffxing, bool to int conversions and output renaming
+
+		network = IECoreScene.ShaderNetwork(
+			shaders = {
+				"blackbodyHandle" : IECoreScene.Shader( "blackbody", "ai:shader", IECore.CompoundData( { "normalize" : IECore.BoolData( True ) } ) ),
```

Comment:
We can omit the explicit `CompoundData/BoolData` here (and likewise elsewhere) and rely on automatic conversion. That can make things a fair bit more readable.

Reply:
Amazing, thanks. I always forget when we get implicit conversion and when we don't. See how you feel about https://github.com/GafferHQ/gaffer/pull/3667/commits/1cc59c10e8f3736892f2893f1f377ef0dad858e9.

Reply:
> I always forget when we get implicit conversion and when we don't.

Yeah, we really need to fix the spots that don't support implicit conversion.